### PR TITLE
[DD-1271] extended filter types

### PR DESCRIPTION
**Pull request by boferri**

### CODE REVIEW

File: yo/app/scripts/directives/filter.js [hunk 1]
```diff
@@ -74,10 +74,21 @@ angular.module('dmpApp')
...
+            {id: 'REGEXP', name: 'regular expression'},
+            {id: 'EQUALS', name: 'equals'},
+            {id: 'NOTEQUALS', name: 'not-equals'}
+        ];
+
+        $scope.defaultFilterType = $scope.filterTypes[1];
+
         $scope.addFilter = function() {
 
+            var filter = schemaParser.fromDomainSchema($scope.dataSchema, true, true, $scope.filterTypes, $scope.defaultFilterType);
```
Comment (knutwalker) [inline yo/app/scripts/directives/filter.js]: The way is see it, `defaultFilterType` and `filterType` are not used in the view, so it needn't be defined on `$scope`.
Also, the filterTypes and the defaultType are directly related to each other, so maybe pass an object of `{types, default}` as one argument to `fromDomainSchema` and there, we only need one if-argument-is-defined check.

Comment (boferri) [inline yo/app/scripts/directives/filter.js]: ok, see my enhancements in the commit below.

**Comment by knutwalker** [comment]
great, LGTM

**Comment by chrode** [comment]
FilterTypes are only available when first clicking the "Add Filter" button. Once the project is reloaded no FilterTypes are selectable and the select box stays empty. This looks broken ATM.

**Comment by boferri** [comment]
ok, will try to fix this.

**Comment by boferri** [comment]
@chrode should work now as expected.

**Pull request merged**